local-tests/test_multiple_restarts.py

```diff
@@ -22,28 +29,34 @@ def printt(s): print(ctime() + ' | ' + s)
                 sudo_account_id=keys[phrases[0]],
                 chain_type='local')
 
+printt('Purging previous chain')
+chain.purge()
+
 chain.set_flags('no-mdns',
                 port=Seq(30334),
                 validator_port=Seq(30343),
                 ws_port=Seq(9944),
                 rpc_port=Seq(9933),
                 unit_creation_delay=200,
-                execution='Native',
-                state_pruning='archive')
+                execution='Native')
 addresses = [n.address() for n in chain]
 validator_addresses = [n.validator_address() for n in chain]
 chain.set_flags(bootnodes=addresses[0])
-chain.set_flags_validator(public_addr=addresses, public_validator_addresses=validator_addresses)
+chain.set_flags_validator(public_addr=addresses,
+                          public_validator_addresses=validator_addresses)
+if state_pruning is not None:
+    chain.set_flags('experimental-pruning', state_pruning=state_pruning,
+                    )
 
 chain.set_flags_validator('validator')
 
 printt('Starting the chain')
 chain.start('aleph')
 
-printt('Waiting for finalization')
-chain.wait_for_finalization(0)
 printt('Waiting for authorities')
 chain.wait_for_authorities()
+printt('Waiting for finalization')
+chain.wait_for_finalization(256)
```

Comment:
Same here: why 256?

Reply:
Because pruning starts in substrate at 256. We want to check it does not prune anything